app/controllers/application_controller.rb

```diff
@@ -4,14 +4,30 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :set_cache_header
 
-  helper_method :editable?
+  helper_method :editable?, :likeable?, :logged_in?, :current_developer
 
   private
 
+  def logged_in?
+    current_developer != nil
+  end
+
+  def current_developer
+    if session[:developer_id]
+      @current_developer ||= User.find(session[:developer_id])
+    else
+      @current_developer = nil
+    end
+  end
+
   def editable?(post)
     current_developer && (current_developer == post.developer || current_developer.admin?)
   end
 
+  def likeable?(post)
+    current_developer && (current_developer != post.developer)
```

Comment:
seems a bit weird to see this helper here, not sure if it'd be better in the post model itself

```ruby
def likeable_by(developer)
  post.author_id != developer.id
end
```

Reply:
We use the likeable? validation at the view level to avoid rendering the heart icon if the post is your own!